File: level_1/자연수 뒤집어 배열로 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int length(long long n)
{
    int len = 0;
    while (n)
    {
        len++;
        n /= 10;
    }
    return len;
}

int *solution(long long n)
{
    int size = length(n);
    int *answer = (int *)malloc(sizeof(int) * size);
    for (int i = 0; i < size; i++)
    {
        answer[i] = n % 10;
        n /= 10;
    }
    return answer;
}

File: level_1/자연수 뒤집어 배열로 만들기.java
class Solution {
    public int[] solution(long n) {
        int count = 0;
        long imsy = n;
        while (imsy > 0) {
            count += 1;
            imsy /= 10;
        }
        int[] answer = new int[count];
        int idx = 0;
        while (n > 0) {
            answer[idx] = (int) (n % 10);
            idx++;
            n /= 10;
        }
        return answer;
    }
}